src/style.rs

```diff
@@ -712,7 +747,9 @@ mod tests {
         let mut buf = Vec::new();
 
         let mut style = ProgressStyle::default_bar();
-        style.format_map.insert("foo", |_| "XXX".into());
+        style
+            .format_map
+            .insert("foo", FormatterVariant::Stateless(|_| "XXX".into()));
 
         style.template = Template::from_str("{foo:5}").unwrap();
         style.format_state(&state, &mut buf, WIDTH);
```

Comment:
Please add a tes for a stateful `ProgressTracker`.

Reply:
Done